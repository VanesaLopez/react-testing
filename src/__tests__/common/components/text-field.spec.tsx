import * as React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { TextField } from '../../../common/components/text-field';

describe('TextField specs', () => {
  it('should render', () => {
    // Arrange
    const props = {
        name: 'textField',
        label: 'label',
        value: 'hola',
        onChange: jest.fn(),
        onBlur: jest.fn()
      };

    // Act
    const { getByText, asFragment } = render(<TextField {...props}/>);

    // Assert
    getByText(props.label);
    expect(asFragment()).toMatchSnapshot();
    
  });
});