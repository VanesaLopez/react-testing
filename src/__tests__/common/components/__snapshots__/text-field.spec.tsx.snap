// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField specs should render 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
      data-shrink="true"
    >
      label
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input"
        type="text"
        value="hola"
      />
    </div>
  </div>
</DocumentFragment>
`;
